pkg/instrumentation/sdk.go

```diff
@@ -82,9 +83,11 @@ func injectCommonSDKConfig(otelinst v1alpha1.Instrumentation, ns corev1.Namespac
 	}
 	idx = getIndexOfEnv(container.Env, envOTELPropagators)
 	if idx == -1 && len(otelinst.Spec.Propagators) > 0 {
+		specialScopes := *(*[]string)((unsafe.Pointer(&otelinst.Spec.Propagators)))
```

Comment:
This now has the potential to break the operator: could you please add some negative tests? Perhaps isolate this in its own func to facilitate testing.

Reply:
> This now has the potential to break the operator

Could you please explain why it has the potential to break the operator?

Reply:
I need to play around, but you have a few conversions and pointer operations happening there. If you are confident that they are safe with all possible types of inputs, then I'm fine with it.

Reply:
I am fine with this.

You can do the exercise and get back here if this is risky.